### PR TITLE
feat: add build.crossorgin

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Review by TheAlexLichter — APPROVED** [review]

**Review by aldarund — APPROVED** [review]

File: packages/server/src/middleware/nuxt.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ export default ({ options, nuxt, renderRoute, resources }) => async function nux
 
       const links = pushAssets
         ? pushAssets(req, res, publicPath, preloadFiles)
-        : defaultPushAssets(preloadFiles, shouldPush, publicPath, options.dev)
+        : defaultPushAssets(preloadFiles, shouldPush, publicPath, options)
```
Comment (pi0) [inline packages/server/src/middleware/nuxt.js]: This change would be a breaking change for `2.x` releases (changing function signature). If the options than `dev` we can add it as another param. For `3.x` i suggest changing signature to accept an object with named args.

Comment (clarkdo) [reply packages/server/src/middleware/nuxt.js]: `defaultPushAssets` is not the one exposed to users, it's only for internal use.
`pushAssets` is for user config, we can add options to it in 3.x.

File: packages/webpack/src/config/client.js [hunk 2]
```diff
@@ -124,6 +125,12 @@ export default class WebpackClientConfig extends WebpackBaseConfig {
       }))
     }
 
+    if (this.options.build.crossorigin) {
+      plugins.push(new CorsPlugin({
```
Comment (manniL) [inline packages/webpack/src/config/client.js]: Could be require the plugin lazily here (Would that make any difference)?

Comment (clarkdo) [reply packages/webpack/src/config/client.js]: This part is for building, so it won't be bundled into server.js, so I thinks it's fine.

**Review by atinux — APPROVED** [review]

**Review by pi0 — APPROVED** [review]

**Pull request merged**